synapse/handlers/message.py

```diff
@@ -496,6 +508,79 @@ def create_event(self, requester, event_dict, token_id=None, txn_id=None,
 
         defer.returnValue((event, context))
 
+    def _is_exempt_from_privacy_policy(self, builder):
+        """"Determine if an event to be sent is exempt from having to consent
+        to the privacy policy
+
+        Args:
+            builder (synapse.events.builder.EventBuilder): event being created
+
+        Returns:
+            Deferred[bool]: true if the event can be sent without the user
+                consenting
+        """
+        # the only thing the user can do is join the server notices room.
+        if builder.type == EventTypes.Member:
+            membership = builder.content.get("membership", None)
+            if membership == Membership.JOIN:
+                return self._is_server_notices_room(builder.room_id)
+        return succeed(False)
```

Comment:
Do we not want to exempt AS users?

Reply:
Oh, yo do that later. I'm not sure I really get the difference between the checks here and the checks later

Reply:
well, it's a difference between "does this *action* require consent to the privacy policy?" and "does this *user* need to consent to the privacy policy?".

the distinction means that it is meaningful to call `assert_accepted_privacy_policy` from the RoomCreationHandler, where we have already established that the action requires consent, but not whether the user needs to do so.